### DISCUSSION TITLE
Error: unsupported workflow run event

### DISCUSSION THREAD

**Discussion by dpo** [community/community#34117] [Actions] [answered]
Hi,

I have a workflow that runs
```yaml
on:
  workflow_run:
    workflows: [bmark_M1]
    types: [completed]
```
but fails immediately with the message
```
Error: Error: unsupported workflow run event
```
https://github.com/JuliaSmoothOptimizers/LimitedLDLFactorizations.jl/actions/runs/3121218878/jobs/5062383075

The workflow in question is supposed to post a link in a pull request to artifacts generated by the [`bmark_M1` workflow](https://github.com/JuliaSmoothOptimizers/LimitedLDLFactorizations.jl/blob/main/.github/workflows/bmark_M1.yml). I'm using this setup in other repos and it works just fine. I've puzzled over why it would fail in this case and I hope someone here has an idea.

The `bmark_M1` workflow runs on a self-hosted server. That's the only difference I can point to. Would that matter?

Thanks in advance!

**Comment by antonblr** [comment, marked as answer]
The error comes from the underlying library used by `veitbjarsch/artifacts-url-comments@v1.1.0` - https://github.com/veitbjarsch/github-actions/blob/7620dbab91e5a20870d4bc8ee6a699c0f3500cae/src/workflow/workflowGetPullRequest.ts#L33

Looking at the code it works for workflows triggered by pull request / code push only. E.g. if you trigger your `bmark_M1` workflow by re-running the downstream `add artifact links to pr` most likely will throw this error.

```ts
switch (payload.workflow_run.event) {
  case 'push':
    return getPullRequestFromCommitMessage(
      octokit,
      payload.workflow_run.head_commit.message
    )
  case 'pull_request':
    return getPullRequestForPullRequestWorkflow(octokit, payload)
  default:
    throw new Error('unsupported workflow run event')
}
```

**Reply by dpo** [reply]
Indeed, thank you! For some reason, typing "unsupported workflow run event" in the GithHub search box on that repository returns nothing, which is why I missed it.

I guess I should either add `pull_request_target` to the list of supported events in that Action, or use a different approach.

Thanks again!
